src/java/voldemort/server/HostMatcher.java

```diff
@@ -0,0 +1,66 @@
+package voldemort.server;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+
+import voldemort.VoldemortApplicationException;
+import voldemort.cluster.Node;
+import voldemort.utils.ReflectUtils;
+
+
+public abstract class HostMatcher {
+
+    private final List<String> types;
```

Comment:
Given comments above, this `types` member and all the code that uses it is completely unnecessary.